src/components/Vote/Vote.tsx

```diff
@@ -0,0 +1,68 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import React from 'react';
+
+import { ComponentPropsWithoutRef } from 'react';
+
+import { TbTriangle, TbTriangleFilled } from 'react-icons/tb';
+
+export interface VoteProps extends ComponentPropsWithoutRef<'div'> {
+  votes: number;
+  voted?: 'VOTE' | 'UNVOTE';
+  onVoteUp: () => void;
+  onVoteDown: () => void;
+  customClass?: string;
+}
+
+export function Vote(props: VoteProps) {
+  const { votes, voted, onVoteUp, onVoteDown, customClass, ...rest } = props;
+
+  function onClickItem(arg: 1 | -1) {
+    switch (voted) {
+      case 'VOTE': {
+        arg === 1 ? onVoteDown() : (onVoteDown(), onVoteDown());
+        break;
+      }
+      case 'UNVOTE': {
+        arg === -1 ? onVoteUp() : (onVoteUp(), onVoteUp());
+        break;
+      }
+      default: {
+        arg === 1 ? onVoteUp() : onVoteDown();
+        break;
+      }
+    }
+  }
+
+  return (
+    <div
+      data-testid="vote"
+      className={`text-f-primary cursor-default ${customClass}`}
+      {...rest}
+    >
+      <div
+        data-testid="vote-up"
+        className="cursor-pointer"
+        onClick={() => onClickItem(1)}
+      >
+        {voted === 'VOTE' ? (
```

Comment:
BTW, extract that for a upper const:

`const isVote = voted === "VOTE"`

then just use here isVote